### PR TITLE
fix when creating an entity it should be added to the context

### PR DESCRIPTION
**Pull request by BerendWouters**

### CODE REVIEW

File: Singer.API/DTOs/Users/CreateCareUserDTO.cs [hunk 1]
```diff
@@ -37,5 +34,14 @@ public class CreateCareUserDTO : CreateUserDTO
 
       [DisplayName("Middelen")]
       public bool HasResources { get; set; }
+
+      [DisplayName("Voornaam")]
```
Comment (wim07101993) [inline Singer.API/DTOs/Users/CreateCareUserDTO.cs:42]: No data validation on any of the properties

File: Singer.API/DTOs/Users/CreateUserDTO.cs [hunk 2]
```diff
@@ -21,6 +21,6 @@ public class CreateUserDTO : ICreateUserDTO
       [Required]
       [EmailAddress]
       [DisplayName("E-mail adres")]
-      public string Email { get; set; }
+      public virtual string Email { get; set; }
```
Comment (wim07101993) [inline Singer.API/DTOs/Users/CreateUserDTO.cs]: is virtual still necessary?

File: Singer.API/Services/UserService.cs [hunk 3]
```diff
@@ -39,6 +39,11 @@ protected UserService(ApplicationDbContext context, IMapper mapper, UserManager<
             Email = dto.Email,
             UserName = dto.Email
          };
+
+         if (string.IsNullOrEmpty(baseUser.UserName))
```
Comment (wim07101993) [inline Singer.API/Services/UserService.cs:45]: maybe comment with reason to generate random name

File: Singer.API/Services/UserService.cs [hunk 4]
```diff
@@ -54,11 +59,17 @@ protected UserService(ApplicationDbContext context, IMapper mapper, UserManager<
...
 
+      private string GenerateRandomUserName(string firstName, string lastName)
+      {
+         var randomGuid = Guid.NewGuid();
+         return $"{firstName}.{lastName}.{randomGuid}@test.com";
+      }
+
       public override async Task<TUserDTO> GetOneAsync(Guid id)
       {
-         var item = await Queryable
```
Comment (wim07101993) [inline Singer.API/Services/UserService.cs:59]: not async?

Comment (BerendWouters) [reply Singer.API/Services/UserService.cs:59]: Automapper caused an issue with IAsyncEnumerable and the SingleOrDefault.

**Review by wim07101993 — APPROVED** [review]

**Pull request merged**